docs: add explanation of p2p configuration options (#5397)
tendermint/tendermint: docs/tendermint-core/configuration.md

- `timeout_commit` = how long we wait after committing a block, before starting
  on the new height (this gives us a chance to receive some more precommits,
  even though we already have +2/3)

## P2P settings 

This section will cover settings within the p2p section of the `config.toml`. 

- `external_address` = is the address that will be advertised for other nodes to use. We recommend setting this field with your public IP and p2p port. 

> - `seeds` = is a list of comma separated seed nodes. A seed node is a node that does not participate in consensus but only helps propagate peers to nodes in the networks

Reviewer: ```suggestion
- `seeds` = is a list of comma separated seed nodes that you will connect upon a start and ask for peers. A seed node is a node that does not participate in consensus but only helps propagate peers to nodes in the networks
```

Author: Is it really only on start? I thought it may check in with the seed here and there? (this was my assumption, so can easily be wrong)

Reviewer: not only on start.. I just wanted to emphasize that you will connect to it in the absence of peers and request more of them.

- `persistent_peers` = is a list of comma separated peers that you will always want to be connected to. If you're already connected to the maximum number of peers, persistent peers will not be added. 
- `max_num_inbound_peers` = is the maximum number of peers you will accept inbound connections from at one time (where they dial your address and initiate the connection).
- `max_num_outbound_peers` = is the maximum number of peers you will initiate outbound connects to at one time (where you dial their address and initiate the connection).

> - `unconditional_peer_ids` = is similar to `persistent_peers` except that these peers will be connected to even if you are already connected to the maximum number of peers.

Reviewer: ```suggestion
- `unconditional_peer_ids` = is similar to `persistent_peers` except that these peers will be connected to even if you are already connected to the maximum number of peers. This can be a validator node ID on your sentry node.
```

- `pex` = turns the peer exchange reactor on or off. If you are running a validator, you will want the `pex` turned off so it would not begin gossiping to unknown peers on the network.

> - `seed_mode` = is used for when node operators want to run their node as a seed node. To minimize the servers usage setting the mempool's size to 0 is recommended.

Reviewer 2: ```suggestion
- `seed_mode` = is used for when node operators want to run their node as a seed node. Seed node's run a variation of the PeX protocol that disconnects from peers after sending them a list of peers to connect to. To minimize the servers usage, it is recommended to set the mempool's size to 0.
```

> -  `private_peer_ids` = is a comma separated list of node ids that you would not like exposed. This can be filled with a validators node id. 

Reviewer 2: ```suggestion
-  `private_peer_ids` = is a comma separated list of node ids that you would not like exposed to other peers (ie. you will not tell other peers about the private_peer_ids). This can be filled with a validators node id. 
```
